reflect.go

```diff
@@ -552,20 +583,24 @@ func (r *Reflector) addDefinition(definitions Definitions, t reflect.Type, s *Sc
 }
 
 // refDefinition will provide a schema with a reference to an existing definition.
-func (r *Reflector) refDefinition(definitions Definitions, t reflect.Type) *Schema {
+func (r *Reflector) refDefinition(_ Definitions, t reflect.Type) *Schema {
 	name := r.typeName(t)
-	if def, ok := definitions[name]; ok {
-		if def.Anchor != "" {
-			return &Schema{
-				Ref: "#" + def.Anchor,
-			}
-		}
-	}
 	return &Schema{
 		Ref: "#/$defs/" + name,
 	}
 }
 
+func (r *Reflector) lookupID(t reflect.Type) ID {
+	if r.Lookup != nil {
+		if t.Kind() == reflect.Ptr {
+			t = t.Elem()
+		}
+		return r.Lookup(t)
```

Comment:
The godoc say we'll never pass a pointer, but I don't think guarantee that here.  If we call `lookupID()` with `t` being a pointer to a pointer (to a pointer... ad infinitum), we'll still pass a pointer to r.Lookup.  We probably need a loop of `t = t.Elem()`, or some other guarantee that I don't yet see further up the call chain.